core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -545,42 +697,101 @@ class ReplicaManager(val config: KafkaConfig,
     Option(allPartitions.get(topicPartition)).getOrElse(HostedPartition.None)
   }
 
-  def isAddingReplica(topicPartition: TopicPartition, replicaId: Int): Boolean = {
+  def isAddingReplica(topicPartition: TopicPartition, replicaId: Int, includeFenced: Boolean): Boolean = {
```

Comment:
As far as I can tell, we don't have any use cases which call for `includeFenced=true`. Did I miss any? Perhaps we should just document that this returns false for fenced partitions.